src/Main.purs

```diff
@@ -151,6 +156,14 @@ handleAction :: forall output. Action -> H.HalogenM State Action ChildSlots outp
 handleAction action = do
   state <- H.get
   case action of
+    Clear -> plotRoughThenRobust state state.bounds (map (\plot -> plot { expression = Nothing }) state.plots)
+    HandleExpressionInput (Parsed id expression text) -> plotRoughThenRobust state state.bounds (map (updatePlot id expression text) state.plots)
+    Pan direction -> plotRoughThenRobust state (panBounds state.bounds direction) state.plots
+    Zoom isZoomIn -> plotRoughThenRobust state (zoomBounds state.bounds isZoomIn) state.plots
+    ResetBounds -> plotRoughThenRobust state initialBounds state.plots
+    HandleCanvas (Dragged delta) -> plotRoughThenRobust state (panBoundsByVector state.input.size state.bounds delta) state.plots
+    AddPlot -> H.put state { plots = state.plots <> [ newPlot (1 + length state.plots) ] }
+    HandleBoundsInput (Updated newBounds) -> plotRoughThenRobust state newBounds state.plots
```

Comment:
Heads-up:  When we do the batched computation, it would be good to plot rough only on those segments where we do not have an enclosure yet.

Reply:
Okie dokie